### DISCUSSION TITLE
Slash in Client ID

### DISCUSSION THREAD

**Discussion by ghost** [emqx/emqx#12832] [Q&A] [answered]
Hi, I've noticed that I can set _Common Name_ in my client's cert, which is mapped to _Client ID_, to a string with a slash:

**zone1/client1**

...and it works for me in _acl.conf_:

**{allow, {client, {re, "^[a-f0-9]{8}$"}}, all, ["nodes/${clientid}/+"]}**

It looks like the "/" is treated as a regular part of the path internally, and allows expected subscriptions:

**nodes/zone1/#
nodes/zone1/+/+**

The question is:

**Can I rely on that behavior? Is it safe for production?**

I like it, but for security reasons, I must keep "zone" as the prefix of the _Client ID_, so I could simply have:

**nodes/zone1.client1**

...but then I end up with a flat structure for all the zones and clients, so I'm losing the benefit of the proper hierarchy.

There seems to be zero documentation addressing my case, so I'd like to collect your opinions - thanks!

**Comment by zmstone** [comment, marked as answer]
Hi @A626621 
Yes, it's OK to use slash in clientid.
It might be a bit inconvenient though, because it requires URI encoding (percent encoding) if you want to call REST APIs.

An alternative solution (will be released in 5.7.0):  https://github.com/emqx/emqx/pull/12750

release note:
https://github.com/emqx/emqx/pull/12750/files#diff-232cf17a84770e1e19e76090ae4b44ed80a4f785652500f50126c5d88c6c0d2c
